### PR TITLE
feat: configure HTTP/2 protocol

### PR DESCRIPTION
**Pull request by kov117**
### Checklist

- [x] I've updated both the `azurerm_linux_function_app` and `azurerm_windows_function_app` resources.

### CODE REVIEW

**Comment by kov117** [comment]
Test pass 🎆

File: variables.tf [hunk 1]
```diff
@@ -216,6 +216,12 @@ variable "identity_ids" {
   default     = []
 }
 
+variable "http2_enabled" {
+  description = "Specifies if the HTTP2 protocol should be enabled."
```
Comment (hknutsen) [inline variables.tf:220]: Make description more similar to existing variables:

```suggestion
  description = "Should the HTTP2 protocol be enabled for this Function App?"
```

**Review by hknutsen — CHANGES_REQUESTED** [review]

**Comment by kov117** [comment]
Test passed 👍🏼

File: tests/configuration.unit.tftest.hcl [hunk 2]
```diff
@@ -111,3 +151,43 @@ run "windows_basic_authentication_enabled" {
...
+    location                   = run.setup_tests.location
+    app_service_plan_id        = run.setup_tests.app_service_plan_id
+    storage_account_id         = run.setup_tests.storage_account_id
+    log_analytics_workspace_id = run.setup_tests.log_analytics_workspace_id
+
+    http2_enabled = false
+  }
+
+    assert {
+    condition = azurerm_linux_function_app.this[0].site_config[0].http2_enabled == false
```
Comment (hknutsen) [inline tests/configuration.unit.tftest.hcl:170]: Typo:

```suggestion
    condition = azurerm_windows_function_app.this[0].site_config[0].http2_enabled == false
```

File: tests/configuration.unit.tftest.hcl [hunk 3]
```diff
@@ -111,3 +151,43 @@ run "windows_basic_authentication_enabled" {
...
+    location                   = run.setup_tests.location
+    app_service_plan_id        = run.setup_tests.app_service_plan_id
+    storage_account_id         = run.setup_tests.storage_account_id
+    log_analytics_workspace_id = run.setup_tests.log_analytics_workspace_id
+
+    http2_enabled = true
+  }
+
+    assert {
+    condition = azurerm_linux_function_app.this[0].site_config[0].http2_enabled == true
```
Comment (hknutsen) [inline tests/configuration.unit.tftest.hcl:190]: Typo:

```suggestion
    condition = azurerm_windows_function_app.this[0].site_config[0].http2_enabled == true
```

**Review by hknutsen — CHANGES_REQUESTED** [review]

**Comment by kov117** [comment]
Terraform test passed 👍🏼

**Review by hknutsen — APPROVED** [review]
Looks good 👍

**Pull request merged**